sdcm/utils/gce_builder.py

```diff
@@ -0,0 +1,239 @@
+            for warning in operation.warnings:
+                print(f" - {warning.code}: {warning.message}", file=sys.stderr, flush=True)
+
+        return result
+
+    def _create_launch_template(self) -> compute_v1.InstanceTemplate:
+        """
+        Create a new instance template with the provided name and a specific
+        instance configuration.
+
+        Returns:
+            InstanceTemplate object that represents the new instance template.
+        """
+        # The template describes the size and source image of the boot disk
+        # to attach to the instance.
+        disk = compute_v1.AttachedDisk()
+        initialize_params = compute_v1.AttachedDiskInitializeParams()
+        initialize_params.source_image = (
+            self.runner.image.extra.get('selfLink')
+        )
+        initialize_params.disk_type = 'pd-standard'
+        initialize_params.disk_size_gb = 80
+        disk.initialize_params = initialize_params
+        disk.auto_delete = True
+        disk.boot = True
+
+        # The template connects the instance to the `default` network,
+        # without specifying a subnetwork.
+        network_interface = compute_v1.NetworkInterface()
+        network_interface.network = self.region.network.self_link
+
+        # The template lets the instance use an external IP address.
+        access_config = compute_v1.AccessConfig()
+        access_config.name = "External NAT"
+        access_config.type_ = "ONE_TO_ONE_NAT"
+        access_config.network_tier = "PREMIUM"
+        network_interface.access_configs = [access_config]
+
+        template = compute_v1.InstanceTemplate()
+        template.name = self.name
+        template.properties.disks = [disk]
+        template.properties.machine_type = "e2-medium"
+        template.properties.network_interfaces = [network_interface]
+
+        metadata = compute_v1.Metadata()
+        metadata.items += [{"key": "RunByUser", "value": "QA"}]
+        metadata.items += [{"key": "NodeType", "value": "builder"}]
+        metadata.items += [{"key": "keep", "value": "alive"}]
+
+        template.properties.metadata = metadata
+        template_client = compute_v1.InstanceTemplatesClient(credentials=self.credentials)
+        try:
+            operation = template_client.insert(
+                project=self.region.project, instance_template_resource=template
+            )
+
+            self._wait_for_extended_operation(operation, "instance template creation")
+        except google.api_core.exceptions.Conflict:
+            pass
+
+        return template_client.get(project=self.region.project, instance_template=self.name)
+
+    def configure(self):
+        self._create_launch_template()
+        self._add_cloud_configuration_to_jenkins()
+
+    @classmethod
+    def configure_in_all_region(cls, regions=None):
+        for project in SUPPORTED_PROJECTS:
+            os.environ['SCT_GCE_PROJECT'] = project
```

Comment:
Are you sure that usage of this env var won't trigger any problem like before with cleanup?
Maybe better to specify it somehow else?

Reply:
it's less prone for this error since it's doing just on specific call across all of the project,
the cleanup had multiple steps that wasn't taking into account.

I would be nice to use the same machinery

but this is a manual action someone would be doing, to update/rebuild the needed configuration, not something happening on each sct job, so the price of failure is lower, and someone would be attending to it straight away

Reply:
fixed